src/libratbag-data.c

```diff
@@ -403,6 +411,7 @@ file_data_matches(struct ratbag *ratbag,
 	_cleanup_(g_strfreevp) char **ledtypes_strv = NULL;
 	_cleanup_(ratbag_device_data_unrefp) struct ratbag_device_data *data = NULL;
 	int rc;
+	char *devicetype;
```

Comment:
Needs a `_cleanup_free_`.

Reply:
..or it doesn't? This causes a segmentation fault. Sorry.

Although `g_key_file_get_string()` is supposed to return a newly allocated string per glib docs, perhaps there is more into it.

Reply:
did you init it to `NULL`? There are some paths where a `_cleanup_free_` would be triggered and cause a free of an uninitialized value - maybe that was the reason? glib definitely returns an allocated string here.

Reply:
Oh, you are right, I missed that!